data-prepper-plugins/elasticsearch/README.md

```diff
@@ -120,35 +87,39 @@ A boolean flag indicates APM trace analytics service map data type. e.g.
   "traceGroupName": "MakePayement.auto"
 }
 ```
-
 Default value is false. 
 
-### Custom data type
+- <a name="index"></a>`index`: A String used as index name for custom data type. Applicable and required only If both `trace_analytics_raw` and `trace_analytics_service_map` are set to false.
 
-If both `trace_analytics_raw` and `trace_analytics_service_map` are set to false, the ES sink will ingest custom data type, i.e.
-user needs to provide custom [index](#index).
+- <a name="template_file"></a>`template_file`(optional): A json file path to be read as index template for custom data ingestion. The json file content should be the json value of
+`"template"` key in the json content of elasticsearch [Index templates API](https://www.elastic.co/guide/en/elasticsearch/reference/7.8/index-templates.html), 
+e.g. [otel-v1-apm-span-index-template.json](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/master/data-prepper-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json)
 
-### <a name="index"></a>Index
+- `dlq_file`(optional): A String of absolute file path for DLQ failed output records. Defaults to null.
+If not provided, failed records will be written into the default log file.
 
-A String used as index name for custom data type.
+- `bulk_size` (optional): A long of bulk size in bulk requests in MB. Default to 5 MB. If set to be less than 0, 
+all the records received from the upstream prepper at a time will be sent as a single bulk request. 
+If a single record turns out to be larger than the set bulk size, it will be sent as a bulk request of a single document.
```

Comment:
Do we have any guidance on when this value should be adjusted? (Or maybe that will be discussed in a separate "tuning" section).

Reply:
I will add additional info based on elasticsearch documents.

Reply:
Hmm...Unfortunately, from https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html: 
```
There is no "correct" number of actions to perform in a single bulk request. Experiment with different settings to find the optimal size for your particular workload.
```
So no meaningful guidance yet.

Reply:
Cool, thanks for checking.